### ISSUE TITLE
Publish previous JSON Schema versions to the JSON Schema Store

### ISSUE THREAD

**Issue by VictorIreri** [detekt/detekt#5895]
## Context
The JSON schema URL, specified in [configurations.md](https://github.com/detekt/detekt/blob/main/website/docs/introduction/configurations.md?plain=1#LL182-L182C29), is now versioned and the docs were recently updated for the latest version in https://github.com/detekt/detekt/pull/5881. However, only the latest JSON schema ([1.22.0](https://json.schemastore.org/detekt-1.22.0.json)) is available in the Schema Store.

## Expected behaviour
1. All supported versions of detekt should have their schemas uploaded to the JSON Schema Store.
2. The JSON schema itself should support multiple versions so that supported IDEs can automatically source and apply the correct one.
    * There is a way to create a parent schema that points to versioned child schemas based on some value that would be specified in `detekt.yml`, but I haven't explored this further. 
    https://github.com/SchemaStore/schemastore/issues/915
3. Document the release strategy for patch versions in [CONTRIBUTING.md](https://github.com/detekt/detekt/blob/main/.github/CONTRIBUTING.md) and consider if releasing patches `1.21.1` & `1.22.1` to update the broken JSON Schema URLs in the versioned website docs is worthwhile.
&nbsp;
v1.21.0
https://github.com/detekt/detekt/blob/f081ebe6f99d9069a3a7aa43e354c4aff4a7bc59/website/versioned_docs/version-1.21.0/introduction/configurations.md?plain=1#L165
v1.22.0
https://github.com/detekt/detekt/blob/f081ebe6f99d9069a3a7aa43e354c4aff4a7bc59/website/versioned_docs/version-1.22.0/introduction/configurations.md?plain=1#L182

## Actual behaviour
Users on detekt versions prior to 1.22.0 are not informed of where to get the JSON schema from since the schemas don't exist in the Schema Store. Also, some IDEs automatically source and apply the latest schema ([1.22.0](https://json.schemastore.org/detekt-1.22.0.json)) regardless of the detekt version in the project which may lead to confusion.

**Comment by cortinico** [comment]
Related to #4162

**Comment by TWiStErRob** [comment]
> Also, some IDEs automatically source and apply the latest schema ([1.22.0](https://json.schemastore.org/detekt-1.22.0.json)) regardless of the detekt version in the project which may lead to confusion.

Probably a feature request for [detekt/detekt-intellij-plugin](https://github.com/detekt/detekt-intellij-plugin)...

**Comment by 3flex** [comment]
> All supported versions of detekt should have their schemas uploaded to the JSON Schema Store

We only support the latest version, so in a way we're already do this :)

Instead of going back to populate old versions we could instead make updating this part of the release process and make sure all future releases are versioned.

**Comment by BraisGabin** [comment]
Do we even publish those? I think that right now we are not doing it.

**Comment by 3flex** [comment]
It's current as there is a schema for 1.22 already available. I'm saying we should update and version the schema with each new release from now on but not go back and provide older schemas, just to limit the scope of work.

**Comment by cortinico** [comment]
> I'm saying we should update and version the schema with each new release from now on but not go back and provide older schemas, just to limit the scope of work.

Agree. If we hook this in the release workflow I can take care of it 👍
